### ISSUE TITLE
Consider boundary loss

### ISSUE THREAD

**Issue by cwerner** [cwerner/deadtrees#65]
The Dice Loss is not a great metric for segmentation/ object detections since it favours large segmentation areas and thus tends to ignore smaller objects. One solution might be to, additionally, include a `boundary loss` term. Using a boundary loss as compared to a region based loss (IoU, Dice) has the advantage that it is unaffected by class imbalance. It does not consider the entire region, but only the boundary!

![Loss Functions Overview](https://github.com/JunMa11/SegLoss/raw/master/test/LossOverview.PNG)
Source: [https://github.com/JunMa11/SegLoss](https://github.com/JunMa11/SegLoss)

### Overview/ discussion of issue and boundary losses:
> - Reinke et al., 2021: Common Limitations of Image Processing Metrics: A Picture Story, https://arxiv.org/abs/2104.05642
> - Ma, J., et al., 2021: Loss odyssey in medical image segmentation. Medical Image Analysis, p.102035., https://doi.org/10.1016/j.media.2021.102035 (code: https://github.com/JunMa11/SegLoss)
> - https://jeune-research.tistory.com/entry/Loss-Functions-for-Image-Segmentation-Distance-Based-Losses
 
### Relevant papers
> - Kervadec et al., 2019: Boundary loss for highly unbalanced segmentation, Proceedings of The 2nd International Conference on Medical Imaging with Deep Learning, in: Proceedings of Machine Learning Research 102:285–296, https://proceedings.mlr.press/v102/kervadec19a.html; https://github.com/LIVIAETS/boundary-loss
> - Karimi et al., 2019: Reducing the Hausdorff Distance in Medical Image Segmentation with Convolutional Neural Networks, https://arxiv.org/abs/1904.10030, (unofficial: https://github.com/PatRyg99/HausdorffLoss)
